### PR TITLE
Add option for gRPC response compression

### PR DESCRIPTION
**Pull request by WeiranFang**

### CODE REVIEW

File: end2end-test-examples/echo-client/README.md [hunk 1]
```diff
@@ -6,18 +6,46 @@ This is a simple client for e2e testing against our Echo service defined in [ech
 ## Example
 
 Client sends out `numRpcs` number of unary requests to `host` sequentially
-with request payload size of `reqSize`, and expected response payload size of `rspSize`.
+with request payload size of `reqSize`, and expected response payload size of `rspSize`:
 
-Run test:
+```sh
+./gradlew run --args="--numRpcs=100 --reqSize=100 --resSize=100 --host=grpc-cloudapi1.googleapis.com"
```
Comment (rven1) [inline end2end-test-examples/echo-client/README.md:12]: Why is resSize needed for an echo server if response size would be the same as the request size? Or is there a way to have a larger response size?

File: end2end-test-examples/echo-client/README.md [hunk 2]
```diff
@@ -6,18 +6,46 @@ This is a simple client for e2e testing against our Echo service defined in [ech
...
+
+`--warmup`: Number of warm-up RPCs to send before the test. Default: 5.
+
+`--reqSize`: Set request payload size in KB.
+
+`--resSize`: Set response payload size in KB.
+
+`--reqComp`: gRPC compression algorithm to use for request.
+
+`--resComp`: gRPC compression algorithm to use for response.
```
Comment (rven1) [inline end2end-test-examples/echo-client/README.md:45]: Spell out the available compression algorithms (probably only gzip)

File: end2end-test-examples/echo-client/README.md (hunk 1 above)
Comment (WeiranFang) [reply end2end-test-examples/echo-client/README.md:12]: Actually you can set the response size to be different number. The server will respect that number and generates a repeated string with that size.

File: end2end-test-examples/echo-client/README.md (hunk 2 above)
Comment (WeiranFang) [reply end2end-test-examples/echo-client/README.md]: done

**Pull request merged**